07.ls_object/ls.rb

```diff
@@ -0,0 +1,90 @@
+#! /usr/bin/env ruby
+
+# frozen_string_literal: true
+
+require 'optparse'
+require 'etc'
+require_relative 'ls_file'
+require_relative 'ls_directory'
+
+COLUMNS_NUMBER = 3
+
+def main
+  option = ARGV.getopts('arl')
+
+  display_files_in_directory('.', option)
+end
+
+def display_files_in_directory(directory_path, option)
+  directory = LsDirectory.new(directory_path, hidden_file_presence: option['a'], reversed_order: option['r'])
+  ls_files = directory.ls_files
+  return if ls_files.empty?
+
+  if option['l']
+    puts "total #{directory.blocks_sum}"
+    print_files_details(ls_files)
+  else
+    print_files(ls_files)
+  end
+end
+
+def print_files(ls_files)
+  rows_number = ls_files.length.ceildiv(COLUMNS_NUMBER)
+  aligned_files = align_array_size(rows_number, ls_files)
+  transposed_files = aligned_files.each_slice(rows_number).to_a.transpose
+
+  file_name_max_length = ls_files.map { |file| file.file_name.length }.max
+  transposed_files.each do |files|
+    files.each { |file| print "#{file&.file_name&.ljust(file_name_max_length)}\s\s" }
```

Comment:
```suggestion
  transposed_files.each do |ls_files|
    ls_files.each { |ls_file| print "#{ls_file&.file_name&.ljust(file_name_max_length)}\s\s" }
```

実体はLsFileクラスなので、Ruby標準のFileクラスと区別する意味で変数名をこうした方が良いと思います

Reply:
確かにそうですね...><
ご指摘ありがとうございます。